File: sys-src/ScholarLens-backend/database/models/author.py
from beanie import Document, Link
from typing import Optional, List, TYPE_CHECKING, ForwardRef
from pydantic import BaseModel, Field
import uuid

if TYPE_CHECKING:
    from .publication import Publication
    from .institut import Institut

class Author(Document):
    id: uuid.UUID = Field(default_factory=uuid.uuid4)
    google_scholar_id: str
    name: str
    institut: Link["Institut"] = Field(...)
    publications: List[Link["Publication"]] = []
    interest: List[str] = []
    email: str
    h_index: int
    i10_index: int
    citation: int
    h_index5y: int
    i10_index5y: int
    citation5y: int
    cited_by_years: List[int] = []

    class Settings:
        name = "authors"

File: sys-src/ScholarLens-frontend/src/api/apiModels/Author.ts
import { Institut } from "./Institut";
import { Publication } from "./Publication";

export interface Author {
  google_scholar_id: string;
  name: string;
  institut: Institut;
  interests: string[];
  email: string;
  citations?: number;
  h_index?: number;
  i10_index?: number;
  h_index5y?: number;
  i10_index5y?: number;
  citation5y?: number;
  publications?: Publication[];
  cites_per_year: Record<string, number>;
}
